dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/internal/domain/event/DdSpanToSpanEventMapper.kt

```diff
@@ -54,15 +55,12 @@ internal class LegacyToSpanEventMapper(
 
     private fun resolveMeta(event: DDSpan): SpanEvent.Meta {
         val networkInfo = networkInfoProvider.getLatestNetworkInfo()
-        val simCarrier = SpanEvent.SimCarrier(
-            id = networkInfo.carrierId.stringOrNullIfSmaller(),
-            name = networkInfo.carrierName
-        )
+        val simCarrier = resolveSimCarrier(networkInfo)
         val networkInfoClient = SpanEvent.Client(
             simCarrier = simCarrier,
-            signalStrength = networkInfo.strength.stringOrNullIfSmaller(Int.MIN_VALUE),
-            downlinkKbps = networkInfo.downKbps.stringOrNullIfSmaller(),
-            uplinkKbps = networkInfo.upKbps.stringOrNullIfSmaller(),
+            signalStrength = networkInfo.strength?.toString(),
```

Comment:
Sorry I didn't notice that earlier, why are those values strings ?

Reply:
they were Strings before when we added them to the `Json` . I guess is because the trace intake only accepts strings. I am not sure but I just followed what we had before.